### PR TITLE
Dogusata/add markdown parser tests

### PR DESCRIPTION
**Pull request by dogusata**
### Added markdown parser e2e and visual regression tests for:

- [x] It should parse and show generic markdown items like bold or italic texts, links, block quotes, so that users can see rich content in answers.
- [x] It should show code blocks with coding language, so that users can see codes in proper syntax.
- [x] It should show code block actions, so that users can take actions for the code inside that block.
- [x] It should show reference highlights inside code blocks, so that users can see the tooltips by hovering them to see the reference information.
- [x] It should show inline code elements, so that users can see code parts inside paragraphs.
- [x] It should show tables, so that users can see content aligned.
- [x] It should show nested list items, so that users can follow the content with hierarchy.

### Not limited with that, also improvements and fixes added mainly related with markdown rendering while writing the markdown parser tests

- Added support for todo list markdown items
- Added support for `input[type="checkbox"]` to `customRenderer` for todo list item addition
- Refactored `animationEnd` listener
  - Fixed async await problems
  - Fixed reliability issues, now it really waits for the animation end
  - Total time spent to listen animation end improved a lot to match almost the live duration for the animations
- Refactored empty element cleanup for `card-body`
- Updated **border bottom animator**
  -  Color is replaced with the new gradient color
  - Border sizing adjusted depending on the real border width
  - To avoid idle animations, instead of adding the whole pseudo element depending on the parent selector, `display` and `animation` value state changes added according the to parent element's class changes
- Updated `markedjs` version to its latest (which contains several fixes and performance improvements)
- Updated `all markdown items` content in example app
- Updated playwright version 
- Fixed example app stuck after stream ends for default result

**New border bottom animator:**
image

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: example/src/samples/sample-all-in-one.md [hunk 1]
```diff
@@ -13,136 +6,66 @@
...
-## Code
+`inline code`.
 
-### Inline Code Example
-Here is some `inline code`.
-
-### Block Code Example
-This is a block of code.
 ```
 no syntax declared
```
[the remark this answers is not in the document]
Comment (dogusata) [reply example/src/samples/sample-all-in-one.md:44]: Nice nit, can also be with 3 or more `~`. Should also do the check maybe with more than three `.

I'll add them on the next test implementation. However, it might also go to much testing a third part library.

File: src/helper/sanitize.ts [hunk 2]
```diff
@@ -139,6 +140,27 @@ export const cleanHtml = (dirty: string): string => {
     allowedTags: [ ...AllowedTags ],
     allowedAttributes: {
       '*': [ ...AllowedAttributes ],
+      input: [ 'type', 'disabled', 'checked' ]
     },
+    transformTags: {
```
Comment (volodkevych) [inline src/helper/sanitize.ts:145]: Why is this required, and why only for disabled checkboxes?

**Review by volodkevych — APPROVED** [review]

File: example/src/samples/sample-all-in-one.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (volodkevych) [reply example/src/samples/sample-all-in-one.md:44]: I remembered 4 spaces because we had bug with it.

File: src/helper/sanitize.ts (hunk 2 above)
Comment (dogusata) [reply src/helper/sanitize.ts:145]: We don't want to enable all kind of input elements through customRenderer which is being passed through this sanitizer. But for giving the option like the todo lists in markdown (which you can also see at the description of this PR) we want to enable checkboxes, if they are in disabled state. To do that we need to use the transformer to check if it matches with the expected attributes. Otherwise we're returning empty. Which seems like the only option in the [sanitize-html documentation](https://www.npmjs.com/package/sanitize-html).

File: example/src/samples/sample-all-in-one.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (dogusata) [reply example/src/samples/sample-all-in-one.md:44]: Exactly, especially for that we'll add them. But not like checking with 150 ` or ~ chars.
